### PR TITLE
Implement interpolate-1 and interpolate-4 in template plugin test

### PR DESCRIPTION
**Pull request by davidsnam-intel**
### Details:
 - *Interpolate-1*
 - *Interpolate-4*

### Tickets:
 - *CVS-63963*
 - *CVS-64078*

### CODE REVIEW

**Review by wilson-seok — APPROVED** [review]

**Review by jade-cho — APPROVED** [review]

File: src/core/reference/include/ngraph/runtime/reference/interpolate.hpp [hunk 1]
```diff
@@ -78,7 +78,8 @@ class GetNearestPixel final {
                     return static_cast<int64_t>(x_original);
                 }
             };
-        default:;
+        default: {
```
Comment (ilyachur) [inline src/core/reference/include/ngraph/runtime/reference/interpolate.hpp:81]: Why do you need this change?

File: src/core/reference/include/ngraph/runtime/reference/interpolate.hpp [hunk 2]
```diff
@@ -574,6 +576,111 @@ void InterpolateEval<T>::nearest_func(const T* input_data, T* out) {
...
 }
 
+static void pad_input_data(const uint8_t* data_ptr,
+                           uint8_t* padded_data_ptr,
+                           size_t type_size,
+                           const ov::Shape& input_shape,
+                           const ov::Shape& padded_input_shape,
+                           const std::vector<size_t>& pads_begin) {
+    NGRAPH_SUPPRESS_DEPRECATED_START
+    CoordinateTransform input_transform(input_shape);
```
Comment (ilyachur) [inline src/core/reference/include/ngraph/runtime/reference/interpolate.hpp:586]: Can we avoid deprecated classes?

File: src/core/reference/include/ngraph/runtime/reference/interpolate.hpp (hunk 1 above)
Comment (davidsnam-intel) [reply src/core/reference/include/ngraph/runtime/reference/interpolate.hpp:81]: This was caught by cpplint, so I changed.

File: src/core/src/op/interpolate.cpp [hunk 3]
```diff
@@ -470,6 +480,7 @@ bool op::v4::Interpolate::has_evaluate() const {
     case ngraph::element::u8:
     case ngraph::element::f16:
     case ngraph::element::f32:
+    case ngraph::element::f64:
```
Comment (wilson-seok) [inline src/core/src/op/interpolate.cpp:483]: Is this f64? Not bf16?

Comment (davidsnam-intel) [reply src/core/src/op/interpolate.cpp]: Modified and updated. Thanks!

File: src/core/reference/include/ngraph/runtime/reference/interpolate.hpp (hunk 2 above)
Comment (davidsnam-intel) [reply src/core/reference/include/ngraph/runtime/reference/interpolate.hpp:584]: There hasn't been found yet the alternative class or ways for `CoordinateTransform`. And currently Interpolate-4 is using that class while many other ops using too, such as _avg_pool, batch_norm, logical_reduction, log_softmax, lrn, max, max_pool, mean, min, quantize, roi_align, softmax, sum_ and so on. For now it would better use that class, as it looks conventional and stable.

Comment (ilyachur) [reply src/core/reference/include/ngraph/runtime/reference/interpolate.hpp:584]: Could you use `CoordinateTransformBasic`?

File: docs/template_plugin/tests/functional/op_reference/interpolate.cpp [hunk 4]
```diff
@@ -0,0 +1,273 @@
+// Copyright (C) 2018-2021 Intel Corporation
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/interpolate.cpp:1]: Please update copyrights

Comment (davidsnam-intel) [reply docs/template_plugin/tests/functional/op_reference/interpolate.cpp]: Updated

File: src/core/reference/include/ngraph/runtime/reference/interpolate.hpp (hunk 2 above)
Comment (davidsnam-intel) [reply src/core/reference/include/ngraph/runtime/reference/interpolate.hpp]: When using `CoordinateTransformBasic`, it produces following error.

> 21:53:06  [2022-01-24 15:52:53,219] [5724] build ie ERROR: ========== ERRORS were found in the output:
21:53:06  [2022-01-24 15:52:53,219] [5724] build ie ERROR: WARNING: --use-feature=2020-resolver no longer has any effect, since it is now the default dependency resolver in pip. This will become an error in pip 21.0.
21:53:06  [2022-01-24 15:52:53,219] [5724] build ie ERROR: C:\j\workspace\private-ci\ie\build-windows-vs2019\b\repos\openvino\src\core\reference\include\ngraph/runtime/reference/interpolate.hpp(593): error C4996: 'ngraph::CoordinateTransformBasic::index': This method is deprecated and will be removed soon.
21:53:06  [2022-01-24 15:52:53,219] [5724] build ie ERROR: C:\j\workspace\private-ci\ie\build-windows-vs2019\b\repos\openvino\src\core\reference\include\ngraph/runtime/reference/interpolate.hpp(594): error C4996: 'ngraph::CoordinateTransformBasic::index': This method is deprecated and will be removed soon.
21:53:06  [2022-01-24 15:52:53,219] [5724] build ie ERROR: C:\j\workspace\private-ci\ie\build-windows-vs2019\b\repos\openvino\src\core\reference\include\ngraph/runtime/reference/interpolate.hpp(593): error C4996: 'ngraph::CoordinateTransformBasic::index': This method is deprecated and will be removed soon.
21:53:06  [2022-01-24 15:52:53,219] [5724] build ie ERROR: C:\j\workspace\private-ci\ie\build-windows-vs2019\b\repos\openvino\src\core\reference\include\ngraph/runtime/reference/interpolate.hpp(594): error C4996: 'ngraph::CoordinateTransformBasic::index': This method is deprecated and will be removed soon.

Comment (davidsnam-intel) [reply src/core/reference/include/ngraph/runtime/reference/interpolate.hpp:584]: I think for now it'd better use `CoordinateTransform` and later when it will be deprecated, when all of the parts using `CoordinateTransform` should be modified, Interpolate-1 and 4 also will be modified.

File: src/core/tests/CMakeLists.txt [hunk 5]
```diff
@@ -461,7 +461,6 @@ endif()
 # TODO: Enable these tests in new infrastructure
 set(MULTI_TEST_SRC
     backend/abc.in.cpp
-    backend/interpolate.in.cpp
```
Comment (ilyachur) [inline src/core/tests/CMakeLists.txt:464]: Please remove backend/interpolate.in.cpp file

Comment (davidsnam-intel) [reply src/core/tests/CMakeLists.txt:464]: Removed!

**Review by ilyachur — APPROVED** [review]
